src/calendar/CalendarMonth.jsx

```diff
@@ -30,11 +24,39 @@ const CalendarMonth = React.createClass({
     onMonthChange: React.PropTypes.func,
     onYearChange: React.PropTypes.func,
     value: CustomPropTypes.momentOrMomentRange,
+    locale: React.PropTypes.string,
+  },
+
+  _setupLocale(locale) {
+    this.moment = moment();
+
+    if (locale !== 'en') {
+      require(`moment/locale/${locale}`);
+    }
+
+    this.moment.locale(locale);
+
+    const lang = this.moment.localeData(locale);
+
+    this.WEEKDAYS = Immutable.List(lang._weekdays).zip(Immutable.List(lang._weekdaysShort));
+    this.MONTHS = Immutable.List(lang._months);
+  },
+
+  componentWillMount() {
+    const { locale } = this.props;
+    this._setupLocale(locale);
+  },
+
+  componentWillReceiveProps(nextProps) {
+    const { locale } = nextProps;
+    if (locale !== this.props.locale) {
+      this._setupLocale(locale);
+    }
```

Comment:
My guess is that on-the-flow is fine performance-wise; we don't save more than a function call.

Having said that, one problem with my patch (#56) is that an error is raised if ``this.props.locale`` points to a locale that has not been loaded. This error is relatively hard to understand, and should be wrapped in a more "pretty" error message; Having a ``setupLocale`` method provides an opportunity for that.